external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -319,8 +320,10 @@ private boolean emitTupleIfNotEmitted(ConsumerRecord<K, V> record) {
                 } else {
                     collector.emit(tuple, msgId);
                 }
-                emitted.add(msgId);
-                numUncommittedOffsets++;
+                if(!consumerAutoCommitMode){//only need to track in none-AutoCommitMode
+                    emitted.add(msgId);
```

Comment:
That is not needed.  If there are no ackers or if acking is disabled for some other reason ```ack``` is called as part of the call to emit.

https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/executor/spout/SpoutOutputCollectorImpl.java#L142